sources/parsers/parsers.ml

```diff
@@ -47,38 +49,36 @@ let register_parser ~lang new_parser =
     end;
   parsers := (lang, new_parser) :: !parsers
 
-let debug_no_parser_for_extension s =
-  if verbose() then
-    eprintf
-      "error: no parser registered for ext %S. Using default lang ...@." s
-
-let debug_no_parser_for_input_format s =
-  if verbose() then
-    eprintf
-      "error: no parser registered for the provided input format %S ?@." s
+let find_parser ext format =
+  try List.assoc ext !parsers
+  with Not_found ->
+    raise (ParserError ("Error: no parser registered for the provided \
+                         input format %S ?@."^format))
+
+let set_output_format f =
+  if Options.infer_output_format () then
```

Comment:
Actually one question: how is the output inference supposed to work ?

Reply:
And why is it relevant in the parsers ?

Reply:
If output format is not set with the `-o` option then we try to infer the output format accordingly with the input format. If `-o` is not set the flag infer_output_format is true. Then we match the format f, if it's a known format, we set the output format to this format f. If not, we print a message for the user to warn him about the fact that the output format is not known. Since this function is called in get_input_parser, the output inference match the input. If no input is set with `i-` option we try to find a format from file extension, this format will be used in both input and output format. Same if `-i` is set. So by default if you set an input format we will print result according this input format.

Reply:
Well, that might be an interesting thing to document somewhere, :)

Reply:
I've documented it a bit more in options. but since it's done in parser i will add some few line in parsers too